src/sonic-yang-mgmt/sonic_yang.py

```diff
@@ -0,0 +1,490 @@
+    input:   module name
+    returns: JSON or XML format of the input yang module schema tree
+    """
+    def get_module_tree(self, module_name, format):
+        result = None
+
+        try:
+            module = self.ctx.get_module(str(module_name))
+        except Exception as e:
+            print("Cound not get module: " + str(module_name))
+            self.fail(e)
+        else:
+            if (module is not None):
+                if (format == "XML"):
+                    #libyang bug with format
+                    result = module.print_mem(ly.LYD_JSON, ly.LYP_FORMAT)
+                else:
+                    result = module.print_mem(ly.LYD_XML, ly.LYP_FORMAT)
+
+        return result
+
+    """
+    validate_data(): validate data tree
+    input:
+           node:   root of the data tree
+           ctx:    context
+    returns:  Exception if failed
+    """
+    def validate_data (self, node=None, ctx=None):
+        if not node:
+            node = self.root
+
+        if not ctx:
+            ctx = self.ctx
+
+        try:
+            rc = node.validate(ly.LYD_OPT_CONFIG, ctx)
+        except Exception as e:
+            self.fail(e)
+
+    """
+    validate_data_tree(): validate the data tree
+    returns: Exception if failed
+    """
+    def validate_data_tree (self):
+        try:
+            self.validate_data(self.root, self.ctx)
+        except Exception as e:
+            print("Failed to validate data tree")
+            self.fail(e)
+
+    """
+    find_parent_node():  find the parent node object
+    input:    data_xpath - xpath of the data node
+    returns:  parent node
+    """
+    def find_parent_node (self, data_xpath):
+        if (self.root is None):
+            print("data not loaded")
+            return None
+        try:
+            node = self.find_data_node(data_xpath)
+        except Exception as e:
+            print("Failed to find data node from xpath: " + str(data_xpath))
+            self.fail(e)
+        else:
+            if node is not None:
+                return node.parent()
+
+        return None
+
+    """
+    get_parent_xpath():  find the parent node xpath
+    input:    data_xpath - xpathof the data node
+    returns:  - xpath of parent node
+              - Exception if error
+    """
+    def get_parent_xpath (self, data_xpath):
+        path=""
+        try:
+            node = self.find_parent_node(data_xpath)
+        except Exception as e:
+            print("Failed to find parent node from xpath: " + str(data_xpath))
+            self.fail(e)
+        else:
+            if (node is not None):
+                path = node.path()
+        return path
+
+    """
+    new_node(): create a new data node in the data tree
+    input:
+           xpath: xpath of the new node
+           value: value of the new node
+    returns:  new Data_Node object if success,  Exception if falied
+    """
+    def new_node(self, xpath, value):
```

Comment:
Need a standard.
Here we load schema into 'yang.Context' and data into root Node, where each node has associated schema node. 
Can we standardize as

- Node implies data node

- Schema node is explicitly called out as "snode"

- APIs that deal with data node, simply says node (e.g. set_node_value, find_nodes, ...)

- APIs that refer/return schema node, explicitly mention schema (e.g. find_schema_node)

Can we drop dnode, data_nodes, ...